src/sidebar/store/test/use-store-test.js

```diff
@@ -16,90 +18,262 @@ const reducer = (state = initialState, action) => {
-    const warnOnce = sinon.stub();
-    $imports.$mock({
-      '../../shared/warn-once': warnOnce,
+    it('does not re-render if the result of `callback(store)` did not change', () => {
+      mount(<TestComponent />);
+      const originalRenderCount = renderCount;
+      act(() => {
+        testStore.dispatch({ type: 'INCREMENT_OTHER' });
+      });
+      assert.equal(renderCount, originalRenderCount);
+    });
+
+    it('warns if the callback always returns a different value', () => {
+      const warnOnce = sinon.stub();
+      $imports.$mock({
+        '../../shared/warn-once': warnOnce,
+      });
+      const BuggyComponent = () => {
+        // The result of the callback is an object with an `aValue` property
+        // which is a new array every time. This causes unnecessary re-renders.
+        useStore(() => ({ aValue: [] }));
+        return null;
+      };
+      mount(<BuggyComponent />);
+      assert.called(warnOnce);
+      assert.match(warnOnce.firstCall.args[0], /changes every time/);
+    });
+
+    it('unsubscribes when the component is unmounted', () => {
+      const unsubscribe = sinon.stub();
+      testStore.subscribe = sinon.stub().returns(unsubscribe);
+
+      const wrapper = mount(<TestComponent />);
+
+      assert.calledOnce(testStore.subscribe);
+      wrapper.unmount();
+      assert.calledOnce(unsubscribe);
     });
-    const BuggyComponent = () => {
-      // The result of the callback is an object with an `aValue` property
-      // which is a new array every time. This causes unnecessary re-renders.
-      useStore(() => ({ aValue: [] }));
-      return null;
-    };
-    mount(<BuggyComponent />);
-    assert.called(warnOnce);
-    assert.match(warnOnce.firstCall.args[0], /changes every time/);
   });
 
-  it('unsubscribes when the component is unmounted', () => {
-    const unsubscribe = sinon.stub();
-    testStore.subscribe = sinon.stub().returns(unsubscribe);
+  describe('useStoreProxy', () => {
+    let store;
+    let renderCount;
+
+    beforeEach(() => {
+      renderCount = 0;
+      store = createStore([thingsModule]);
+
+      store.addThing('foo');
+      store.addThing('bar');
+
+      $imports.$mock({
+        '../util/service-context': {
+          useService: name => (name === 'store' ? store : null),
+        },
+      });
+    });
+
+    function renderTestComponent() {
+      let proxy;
+
+      const TestComponent = () => {
+        ++renderCount;
+        proxy = useStoreProxy();
+
+        return <div>{proxy.thingCount()}</div>;
+      };
+
+      const wrapper = mount(<TestComponent />);
+      return { proxy, wrapper };
+    }
+
+    it('returns a proxy for the store', () => {
+      const addThingSpy = sinon.spy(store, 'addThing');
 
-    const wrapper = mount(<TestComponent />);
+      const { proxy } = renderTestComponent();
 
-    assert.calledOnce(testStore.subscribe);
-    wrapper.unmount();
-    assert.calledOnce(unsubscribe);
+      assert.ok(proxy);
```

Comment:
Can you elaborate on what this is testing?

The subsequent tests—I think—wouldn't pass if this didn't pass, right?

Reply:
This is just a sanity check that `useStoreProxy` returned something and that the test helper returned that value. It isn't that important.